python/plugins/processing/algs/qgis/RasterLayerStatistics.py

```diff
@@ -58,16 +58,22 @@ def __init__(self):
         super().__init__()
 
     def initAlgorithm(self, config=None):
-        self.addParameter(ParameterRaster(self.INPUT, self.tr('Input layer')))
-
-        self.addOutput(OutputHTML(self.OUTPUT_HTML_FILE, self.tr('Statistics')))
-        self.addOutput(OutputNumber(self.MIN, self.tr('Minimum value')))
-        self.addOutput(OutputNumber(self.MAX, self.tr('Maximum value')))
-        self.addOutput(OutputNumber(self.SUM, self.tr('Sum')))
-        self.addOutput(OutputNumber(self.MEAN, self.tr('Mean value')))
-        self.addOutput(OutputNumber(self.COUNT, self.tr('valid cells count')))
-        self.addOutput(OutputNumber(self.COUNT, self.tr('No-data cells count')))
-        self.addOutput(OutputNumber(self.STD_DEV, self.tr('Standard deviation')))
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT,
+                                                            self.tr('Input layer')))
+        self.addParameter(QgsProcessingParameterNumber(self.BAND,
```

Comment:
I've been thinking - should we have a proper dedicated parameter for raster band selection? (with parent raster layer parameter). We could use the new QgsRasterBandComboBox as the widget, which would then solve https://issues.qgis.org/issues/16047 for processing.